backends/candle/src/models/debertav2.rs

```diff
@@ -0,0 +1,1490 @@
+            let mut position_ids = Vec::with_capacity(elems);
+            let mut attention_mask = Vec::with_capacity(elems);
+            let mut input_lengths = Vec::with_capacity(batch_size);
+            // Bool to know if we need to use the attention mask
+            let mut masking = false;
+
+            for i in 0..batch_size {
+                let start = batch.cumulative_seq_lengths[i] as usize;
+                let end = batch.cumulative_seq_lengths[i + 1] as usize;
+                let seq_length = (end - start) as u32;
+                input_lengths.push(seq_length as f32);
+
+                // Copy values
+                for j in start..end {
+                    input_ids.push(batch.input_ids[j]);
+                    token_type_ids.push(batch.token_type_ids[j]);
+                    position_ids.push(batch.position_ids[j]);
+                    attention_mask.push(1.0_f32);
+                }
+
+                // Add padding if needed
+                let padding = batch.max_length - seq_length;
+                if padding > 0 {
+                    // Set bool to use attention mask
+                    masking = true;
+                    for _ in 0..padding {
+                        input_ids.push(0);
+                        token_type_ids.push(0);
+                        position_ids.push(0);
+                        attention_mask.push(0.0_f32);
+                    }
+                }
+            }
+
+            let attention_mask = match masking {
+                true => {
+                    let attention_mask = Tensor::from_vec(
+                        attention_mask,
+                        (batch_size, max_length, 1),
+                        &self.device,
+                    )?
+                    .to_dtype(self.dtype)?;
+
+                    Some(attention_mask)
+                }
+                false => None,
+            };
+
+            (input_ids, token_type_ids, position_ids, input_lengths, attention_mask)
+        } else {
+            (
+                batch.input_ids,
+                batch.token_type_ids,
+                batch.position_ids,
+                vec![batch.max_length as f32],
+                None,
+            )
+        };
+
+        let input_ids = Tensor::from_vec(input_ids, shape, &self.device)?;
+        let token_type_ids = Tensor::from_vec(token_type_ids, shape, &self.device)?;
+        let position_ids = Tensor::from_vec(position_ids, shape, &self.device)?;
+        let mut input_lengths =
+            Tensor::from_vec(input_lengths, (batch_size, 1), &self.device)?.to_dtype(self.dtype)?;
+
+        let embedding_output = self.embeddings.forward(
+            Some(&input_ids),
+            Some(&token_type_ids),
+            Some(&position_ids),
+            attention_mask.as_ref(),
+            None,
+        )?;
+
+        let encoder_attention_mask = attention_mask.as_ref().cloned().unwrap_or_else(|| Tensor::ones(shape, DType::I64, &self.device).unwrap());
```

Comment:
How about creating the attention mask when batch_size is 1, at line 1202, if we eventually need one?

And it seems like the dtype of the attention mask is float32 when batched input, while int64 in the single input case.